pkg/config/keys.go

```diff
@@ -21,9 +22,15 @@ func MakeZoneKeyPrefix(id SystemTenantObjectID) roachpb.Key {
 	return keys.SystemSQLCodec.ZoneKeyPrefix(uint32(id))
 }
 
-// MakeZoneKey returns the key for id's entry in the system.zones table.
-func MakeZoneKey(id SystemTenantObjectID) roachpb.Key {
-	return keys.SystemSQLCodec.ZoneKey(uint32(id))
+// MakeZoneKey returns the key for a given id's entry in the system.zones table.
+func MakeZoneKey(codec keys.SQLCodec, id descpb.ID) roachpb.Key {
+	return codec.ZoneKey(uint32(id))
+}
+
+// MakeZoneKeyForSystemTenant returns the key for the given system tenant ID's
+// entry in the system.zones table.
+func MakeZoneKeyForSystemTenant(id SystemTenantObjectID) roachpb.Key {
```

Comment:
Let's get rid of this `config.MakeZoneKeyForSystemTenant` API, `MakeZoneKey` is sufficient. For callers that want to only address the SystemTenant, lets have them thread in `SystemSQLCodec`.